src/test/scala/edu/ie3/simona/model/thermal/CylindricalThermalStorageSpec.scala

```diff
@@ -0,0 +1,273 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.thermal
+
+import edu.ie3.datamodel.models.StandardUnits
+import edu.ie3.datamodel.models.input.thermal.CylindricalStorageInput
+import edu.ie3.simona.test.common.UnitSpec
+import edu.ie3.util.quantities.PowerSystemUnits
+import edu.ie3.util.scala.quantities.KilowattHoursPerKelvinCubicMeters
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.matchers.should.Matchers
+import squants.Energy
+import squants.energy.{KilowattHours, Kilowatts}
+import squants.space.CubicMeters
+import squants.thermal.Celsius
+import tech.units.indriya.quantity.Quantities.getQuantity
+import tech.units.indriya.unit.Units
+
+import java.util.UUID
+
+class CylindricalThermalStorageSpec
+    extends UnitSpec
+    with Matchers
+    with BeforeAndAfterAll {
+
+  final val tolerance: Energy = KilowattHours(1e-10)
+
+  var storageInput: CylindricalStorageInput = _
+
+  override def beforeAll(): Unit = {
+    storageInput = new CylindricalStorageInput(
+      UUID.randomUUID(),
+      "ThermalStorage",
+      null,
+      getQuantity(100, StandardUnits.VOLUME),
+      getQuantity(20, StandardUnits.VOLUME),
+      getQuantity(30, StandardUnits.TEMPERATURE),
+      getQuantity(40, StandardUnits.TEMPERATURE),
+      getQuantity(1.15, StandardUnits.SPECIFIC_HEAT_CAPACITY),
+    )
+  }
+
+  def buildThermalStorage(
+      storageInput: CylindricalStorageInput,
+      volume: Double,
```

Comment:
This is absolutely fine and it was like this in the groovy tests. But maybe we could (and should) use the units here. Same below for `vol2Energy`. This would make it a bit safer to use theses tests because one has not to take care that the prefix of the double is correct.